src/Model/Contact.php

```diff
@@ -1933,15 +1931,18 @@ private static function updateFromProbeArray(int $id, array $ret, bool $force =
 		}
 
 		if (!empty($ret['photo']) && ($ret['network'] != Protocol::FEED)) {
-			self::updateAvatar($id, $ret['photo'], $update || $force);
+			self::updateAvatar($id, $ret['photo'], $update);
 		}
 
 		if (!$update) {
 			self::updateContact($id, $uid, $ret['url'], ['failed' => false, 'last-update' => $updated, 'success_update' => $updated]);
 
 			// Update the public contact
 			if ($uid != 0) {
-				self::updateFromProbeByURL($ret['url']);
+				$contact = self::getByURL($ret['url'], false, ['id']);
```

Comment:
Wouldn't `self::getIdByUrl` work here?

Reply:
This is a shortcut especially when you don't want to cause some update, probing or similar. That function has some really weird beaviour and I want to clean it up at some point.

Reply:
Got it.